services/filemanager/s3manager.go

```diff
@@ -190,27 +191,28 @@ func (manager *S3Manager) getSession(ctx context.Context) (*session.Session, err
 	} else {
 		region = *manager.Config.Region
 	}
+
 	var sess *session.Session
-	if manager.Config.AccessKeyID == "" || manager.Config.AccessKey == "" {
-		pkgLogger.Debug("Credentials not found in the destination's config. Using the host credentials instead")
-		sess = session.Must(session.NewSession(&aws.Config{
-			Region:                        aws.String(region),
-			CredentialsChainVerboseErrors: aws.Bool(true),
-			Endpoint:                      manager.Config.Endpoint,
-			S3ForcePathStyle:              manager.Config.S3ForcePathStyle,
-			DisableSSL:                    manager.Config.DisableSSL,
-		}))
-	} else {
-		pkgLogger.Debug("Credentials found in the destination's config.")
-		sess = session.Must(session.NewSession(&aws.Config{
-			Region:                        aws.String(region),
-			Credentials:                   credentials.NewStaticCredentials(manager.Config.AccessKeyID, manager.Config.AccessKey, ""),
-			CredentialsChainVerboseErrors: aws.Bool(true),
-			Endpoint:                      manager.Config.Endpoint,
-			S3ForcePathStyle:              manager.Config.S3ForcePathStyle,
-			DisableSSL:                    manager.Config.DisableSSL,
-		}))
+	awsConfig := aws.Config{
+		Region:                        aws.String(region),
+		CredentialsChainVerboseErrors: aws.Bool(true),
+		Endpoint:                      manager.Config.Endpoint,
+		S3ForcePathStyle:              manager.Config.S3ForcePathStyle,
+		DisableSSL:                    manager.Config.DisableSSL, // TODO: get rid of this options as its risky
+	}
+	if manager.Config.AccessKey != "" && manager.Config.AccessKeyID != "" {
+		pkgLogger.Debug("Credentials found in the destination's config")
+		awsConfig.Credentials = credentials.NewStaticCredentials(manager.Config.AccessKeyID, manager.Config.AccessKey, "")
+	} else if manager.Config.IAMRoleARN != "" {
+		pkgLogger.Debugf("Assuming - %s found in the filemanager config", manager.Config.IAMRoleARN)
```

Comment:
Remove this log as well? (we are logging a part of config)

Reply:
Yes, we are logging the customers ARN but that should not be a secret. That should be ok I think, WDYT?